### DISCUSSION TITLE
Is there a rule to ban lazy built-in imports?

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#6615] [Q&A] [answered]
```python
def read_foo():
    import os  # os should be moved to the top-level.

    return os.getenv("FOO")
```

**Comment by zanieb** [comment]
I don't think we have a rule that covers this — seems worth adding to me.

**Reply by harupy** [reply]
Thanks for the comment! I'm wondering if there is a case where lazy built-in imports live above make sense.

**Reply by charliermarsh** [reply, marked as answer]
We have this: https://github.com/astral-sh/ruff/pull/5180

**Reply by harupy** [reply]
@charliermarsh It looks like C0415 bans any lazy imports:

```
> cat a.py
def f():
    import os
    import pandas

    os.getenv("FOO")


> pylint a.py
************* Module a
a.py:2:4: C0415: Import outside toplevel (os) (import-outside-toplevel)
a.py:3:4: C0415: Import outside toplevel (pandas) (import-outside-toplevel)

------------------------------------------------------------------
Your code has been rated at 2.50/10 (previous run: 2.50/10, +0.00)
```

**Reply by zanieb** [reply]
Ah silly me not checking in-progress pull requests. I'd be interested in a configuration flag for that setting to only include standard library imports. Far too often other imports must be lazy.

**Reply by harupy** [reply]
+1 to a config flag.
